### ISSUE TITLE
Item Damage Incorrectly Converted

### ISSUE THREAD

**Issue by figment** [CleverRaven/Cataclysm-DDA#17958]
Item damage incorrectly calculated on Release x64 build using Visual Studio 2015 u3.
In item.cpp:
```
int item::damage() const {
    return damage_ < 0 ? floor( damage_ ) : ceil( damage_ );
}
```
should be
```
int item::damage() const {
    return static_cast<int>( damage_ < 0 ? floor( damage_ ) : ceil( damage_ ) );
}
```

It would appear that when the double cast to 32-bit int is inlined in this way, the msvc compiler may just effectively return the lower 32-bits of the double which is a really big number and Bash and Cut values for armor are showing wrong values.

Edit:  Looks like this is not a complete fix.  It sometimes works for me some times not.  However the incorrect calculation still appears.  Will try to disable optimizations to fix for now.

#pragma optimize( "", off )  /  #pragma optimize( "", on )  does seem to fix it better but shouldn't really be required.

**Comment by DLaboratory** [comment]
So #17791 I debugged all the day is caused by this?

**Comment by mugling** [comment]
@DLaboratory do you have the vehicle light bug? If so could you please try that `'#pragma` switch and see if this is the root cause?

**Comment by DLaboratory** [comment]
Oh sorry I don't have access to that Windows machine now. I'm using MacBook currently.

**Comment by figment** [comment]
This looks like the same bug.  When I removed optimization I did so around cut_resist/bash_resist as doing so on damage didn't help enough. If I have time I'll try to understand root cause.  I'm currently leaning towards simd instructions like cvttsd2si but need to actually walk through it when happening again.

I recommend closing as duplicate as I did not seem the previous ticket #17791 and they seem the same.
